internal/db/fetcher/fetcher.go

```diff
@@ -281,17 +281,17 @@ func (df *DocumentFetcher) start(ctx context.Context, spans []core.Span, withDel
 			if withDeleted {
 				// DocumentFetcher only ever recieves document keys
 				//nolint:forcetypeassert
-				valueSpans[i] = core.NewSpan(
-					span.Start.(keys.DataStoreKey).WithDeletedFlag(),
-					span.End.(keys.DataStoreKey).WithDeletedFlag(),
-				)
+				span.Start = span.Start.(keys.DataStoreKey).WithDeletedFlag()
+				// The end key should always be the prefix end of the start key
+				span.End = span.Start.PrefixEnd()
+				valueSpans[i] = span
 			} else {
 				// DocumentFetcher only ever recieves document keys
 				//nolint:forcetypeassert
-				valueSpans[i] = core.NewSpan(
-					span.Start.(keys.DataStoreKey).WithValueFlag(),
-					span.End.(keys.DataStoreKey).WithValueFlag(),
-				)
+				span.Start = span.Start.(keys.DataStoreKey).WithValueFlag()
+				// The end key should always be the prefix end of the start key
+				span.End = span.Start.PrefixEnd()
```

Comment:
thought: I think this is incorrect if the user provides two docIDs that are next to each other, there is a function that is called earlier that merges spans that sit next to each other - the change here will mean that the second doc would not be fetched.

e.g.:
```
User (docIDs: [baefooooooa, baefoooooob]{
...
}
```
We currently have no integration tests for this as it is quite painful to find docs with docIds next to eachother, but there are unit tests for the func responsible IIRC.

Reply:
we should create a test action `testUtils.MineDocWithID` :D

Reply:
If you're talking about `core.MergeAscending`, it's called within the same function after this change to the span and should have no impact when querying two adjacent docs.